components/next_steps_view/steps/complete_profile_step/complete_profile_step.tsx

```diff
@@ -0,0 +1,173 @@
+import * as Utils from 'utils/utils';
+
+import {StepComponentProps} from '../../steps';
+
+import './complete_profile_step.scss';
+
+type Props = StepComponentProps & {
+    maxFileSize: number;
+    actions: {
+        updateMe: (user: UserProfile) => void;
+        setDefaultProfileImage: (userId: string) => void;
+        uploadProfileImage: (userId: string, imageData: File) => void;
+    };
+};
+
+type State = {
+    fullName: string;
+    fullNameError?: string;
+    profilePicture?: File;
+    profilePictureError: boolean;
+    removeProfilePicture: boolean;
+};
+
+export default class CompleteProfileStep extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        const user = props.currentUser;
+
+        this.state = {
+            fullName: (user.first_name || user.last_name) ? `${user.first_name} ${user.last_name}` : '',
+            profilePictureError: false,
+            removeProfilePicture: false,
+        };
+    }
+
+    private handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        let fullNameError;
+        if (!event.target.value) {
+            fullNameError = Utils.localizeMessage('next_steps_view.complete_profile_step.nameCannotBeBlank', 'Your name can’t be blank');
+        }
+
+        this.setState({fullName: event.target.value, fullNameError});
+    }
+
+    isFinishDisabled = () => {
+        return Boolean(!this.state.fullName || this.state.fullNameError || this.state.profilePictureError);
+    }
+
+    onSkip = () => {
+        this.props.onSkip(this.props.id);
+    }
+
+    onFinish = () => {
+        const splitName = this.state.fullName.split(' ');
+        const user = Object.assign({}, this.props.currentUser, {first_name: splitName[0], last_name: splitName.slice(1).join(' ')});
+
+        this.props.actions.updateMe(user);
+
+        if (this.state.profilePicture) {
+            this.props.actions.uploadProfileImage(this.props.currentUser.id, this.state.profilePicture);
+        } else if (this.state.removeProfilePicture) {
+            this.props.actions.setDefaultProfileImage(this.props.currentUser.id);
+        }
+
+        this.props.onFinish(this.props.id);
+    }
+
+    onSelectPicture = (profilePicture: File) => {
+        if (!AcceptedProfileImageTypes.includes(profilePicture.type) || profilePicture.size > this.props.maxFileSize) {
+            this.setState({profilePictureError: true});
+            return;
+        }
+
+        this.setState({profilePicture, profilePictureError: false, removeProfilePicture: false});
+    }
+
+    onRemovePicture = () => {
+        this.setState({profilePicture: undefined, profilePictureError: false, removeProfilePicture: true});
+    }
+
+    render() {
+        const {currentUser} = this.props;
+
+        // Make sure picture has been set
+        const pictureSrc = currentUser.last_picture_update ? Utils.imageURLForUser(currentUser.id, currentUser.last_picture_update) : undefined;
```

Comment:
Setting things to undefined can be kind of weird sometimes. I know its not usually a problem, but could this be null?

Reply:
In TypeScript, when defining an optional parameter, it prefers to use `undefined` as the fallback type instead of `null`.
So when you defined a parameter like this: `param?: string`, it's type evaluates to `string | undefined`.
We use `undefined` to match the type.

Reply:
I still have my JavaScript glasses on 🤓